### ISSUE TITLE
Exclamation point is not considered a valid symbol for function name

### ISSUE THREAD

**Issue by Skydev0h** [ton-blockchain/intellij-ton#156]
The highlighter considers that `!` cannot be used in function name.

![image](https://github.com/ton-blockchain/intellij-ton/assets/5331782/e0ab2a3f-8f31-4fcb-b4d2-8eae57362903)

While in reality it is perfectly fine and the code works.

Using the latest v2.0.4 plugin.

**Comment by andreypfau** [comment]
Thank you for report, Will be fixed in next release

**Comment by andreypfau** [comment]
I encountered a problem due to which I had to change the behavior of the lexer and parser, namely how the AST tree is built for ModifyExpression `a~foo()` and the usual ApplyExpression `a foo()`, `foo()`. Because of this, we will have to do a global refactoring of the code base so that there is no degradation in the functionality of current features in the IDE. It can take some time.

But I hope to finish this soon, and the behavior of identifiers will be exactly the same as in the original FunC compiler. Thank you and stay tuned
